### PR TITLE
PHP8 support

### PR DESCRIPTION
**Pull request by AydinHassan**
Hello,

I updated a few things to get the tests running on PHP. The tests are also running on a project which consumes this lib on PHP8 too, which failed before these changes.

There is 1 change I'm not too sure about though - the removal of the bitwise AND in the Token class. Could you maybe shed some light on the intention of the original code?

Thanks!

### CODE REVIEW

**Comment by kadet1090** [comment]
Hi!

Thanks for that submission! Everything seems to be ok.

>There is 1 change I'm not too sure about though - the removal of the bitwise AND in the Token class. Could you maybe shed some light on the intention of the original code?

It's not bitwise AND it's assignment by reference, but I don't really remember why it was done like that - maybe it was some kind of micro optimization for PHP 5.6 as `Token` objects are created in really large quantities and every bit of optimization matters here. Or maybe just leftover of some idea that was not mainstreamed. However I've isolated that change and done benchmarks and cannot see any difference at all, all tests are also good so I don't see any reason why it should be here.

It'd be nice however if you could align `=` signs - I will gladly merge that then :)

File: Parser/Token/Token.php [hunk 1]
```diff
@@ -69,7 +69,7 @@ public function __construct($name = null, array $options = [])
 
         $this->name     = $name;
         $this->rule     = isset($options['rule']) ? $options['rule'] : new Rule();
-        $this->options &= $options;
+        $this->options = $options;
```
Comment (kadet1090) [inline Parser/Token/Token.php:72]: It'd be nice if all `=` signs are aligned (as now they are)

```suggestion
        $this->options  = $options;
```

Comment (AydinHassan) [reply Parser/Token/Token.php:72]: Sure will sort that :)

**Comment by AydinHassan** [comment]
Thanks for the explanation! The reason I asked was because PHP manual said that was a bitwise OP, if you search `$a &= $b` on https://www.php.net/manual/en/language.operators.assignment.php it seems to state that it's a Bitwise Assignment Operators.

**Comment by AydinHassan** [comment]
Thanks @kadet1090 any chance you could ping me when you push a new release out?

**Comment by kadet1090** [comment]
@AydinHassan unfortunately I'm not planning release of 0.9 anytime soon because I don't really have time to finish some things up. However, I can see two solutions:
 - You can use dev-master version (which I'm personally using in every project that uses keylighter), as it is mostly stable... 
 - ...or I can cherry pick that PR to 0.8 branch and release version v0.8.4 - you won't have any features that are currently master only (please see https://github.com/kadet1090/KeyLighter/blob/master/changelog.md) - but you will have stable(ish) version in your composer.json

**Comment by AydinHassan** [comment]
@kadet1090 I'd prefer the second option if it's not too much bother. We are using 0.8.3 currently in a library and to use `dev-master` I'd have to require the dep in every project which consumes the library because of the dev stability.

**Pull request merged**